### DISCUSSION TITLE
Grant user more control around auto-update/auto-install features

### DISCUSSION THREAD

**Discussion by shinebayar-g** [zed-industries/zed#24049] [Feature Requests]
### Summary

While the auto update/download features are convenient they are not really user friendly, consuming unnecessary data.

This applies to both downloading updates and tools (e.g. LSP), as well as the checks for whether Zed & installed tools are out of date. 

In the future it would be beneficial to allow the user more control over disabling/preventing the checking for and auto downloading of updates and tools so that Zed doesn't consume network bandwidth that may be expensive for the user or slow connection networks.

**Comment by Moshyfawn** [comment]
You can set `"auto_update": false` to prevent zed from automatically updating itself.

**Comment by Moshyfawn** [comment]
For Zed reusing the already installed language servers, see #4978 +👍.

**Comment by shinebayar-g** [comment]
![image](https://github.com/zed-industries/zed/assets/3091558/73c1569f-7ce5-4db2-a42e-a2c8a1640855)

Check for updates != Download updates right? I assume checking for updates means, check if there is an update, but not download it right away. They could to be 2 separate actions.

**Comment by probably-neb** [comment]
This seems like a very reasonable concern.

We've decided to start moving enhancements/feature requests like this one to discussions rather than intermixing them with bugs & issues so I'm going to move this to discussions in order to keep tracking this

**Reply by tredondo** [reply]
Why doesn't `auto_update` show up in the suggestions for `auto` when editing `settings.json`? #55514

**Comment by AngularIsAwfull** [comment]
I would love to have auto update on Zed start up and then a voice "update" in Zed menu. I really don't like that my editor is blocked downloading stuff when I'm working on it.

**Reply by probably-neb** [reply]
Are you referring to downloading updates? If I'm understanding your message correctly that's already what we do? Download the update in the background and give you the option to restart and apply the update.

**Reply by AngularIsAwfull** [reply]
Yes, I'm referring to updates. And, yes, I know that updates are being downloaded on start up but not only at this moment! The problem is that: I don't want my editor to self update while I'm working on it unless I specifically want it.

**Reply by quinncomendant** [reply]
Same here. I'm tethering from personal hotspot while traveling and the Zed update killed 375 MB of my 500 MB data plan. 😑
